lib/statesman.rb

```diff
@@ -26,9 +26,14 @@ module Adapters
   def self.configure(&block)
     config = Config.new(block)
     @storage_adapter = config.adapter_class
+    @requires_new = config.requires_new
   end
 
   def self.storage_adapter
     @storage_adapter || Adapters::Memory
   end
+
+  def self.requires_new
+    @requires_new || true
+  end
```

Comment:
Would be great to have some documentation on what this config setting does and why!

Reply:
Would it not be more sensible for it to fallback to `false`?